src/App.tsx

```diff
@@ -1,54 +1,90 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [choosenPerson, setChoosenPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setApliedQuery] = useState('');
+  const [isImputFocus, setIsImputFocus] = useState(false);
+
+  const applyQwery = useCallback(
+    debounce(setApliedQuery, 1000),
+    [],
+  );
+
+  const handleQweryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
The function name 'handleQweryChange' seems to have a typo, it should be 'handleQueryChange'. Proper naming conventions are important for code readability and maintenance.